test/test_common/utility.cc

```diff
@@ -408,19 +408,13 @@ class TimeSystemProvider {
 
 class TestImpl : public TimeSystemProvider, public Impl {
 public:
-  TestImpl(std::chrono::milliseconds file_flush_interval_msec,
-           Thread::ThreadFactory& thread_factory, Stats::Store& stats_store)
```

Comment:
That stats_store was previously needed for the stats  kept by the file-system. Do we not need that anymore?

Reply:
See also #5900 which eliminates that arg when the store is not needed.

Reply:
The API doesn't need the stats_store anymore since it was only ever used by the `Filesystem::File` object, which is now in `AccessLog::AccessLogFile`

Reply:
I see...that's now in AccessLogManager. Don't we want the API to provide the AccessLogManager though, and thus still need access to the StatsStore?

Reply:
Answering myself: the AccessLogManager was previously existing and is owned by the server, which has access to the StatsStore, so it's not needed for that.

I also have to admit that I was planning (but did not document) to more broadly require access to StatsStore especially at initialization time, as part of my long-ongoing #4980 . So if you remove StatsStore from a bunch of call-sites in this PR I'm going to need to re-add it later. I'm OK with that, but I guess I'd say at the risk of speculative generality that providing access to the StatsStore is a reasonable core competency for the API object :)

Reply:
I agree that access to the StatsStore is reasonable functionality for the API object. While it feels a little odd to pass in an object to the API that won't be used, I think it's probably OK if we expect it to be used in the future

Reply:
We added back the StatsStore argument to the Api constructor -- it's just ignored for now